heron/stmgr/src/cpp/server/stmgr-main.cpp

```diff
@@ -48,6 +48,12 @@ DEFINE_string(ckptmgr_id, "", "The id of the local ckptmgr");
 DEFINE_int32(ckptmgr_port, 0, "The port of the local ckptmgr");
 DEFINE_string(metricscachemgr_mode, "disabled", "MetricsCacheMgr mode, default `disabled`");
 
+EventLoopImpl ss;
+
+void sigtermHandler(int signum) {
+  ss.loopExit();
```

Comment:
I am a bit concerned that if loopExit() is safe to call here.
- which thread it is running on and if there might be racing condition.
- if it is possible that loopExit() might get stuck.

Reply:
I think the sigtermHandler should run the in the same thread as any other callback function and loopExit() directly calls event_base_loopbreak() which I think should be safe to call. [Here](http://www.wangafu.net/~nickm/libevent-book/Ref3_eventloop.html) in the libevent book they call event_base_loopbreak() in a very similar way.

But maybe someone with more experience with livevent could give some input on that.